packages/common/src/ContractInteractor.ts

```diff
@@ -308,6 +309,55 @@ export class ContractInteractor {
     }
   }
 
+  async _validateERC165InterfacesRelay (): Promise<void> {
+    const allNull = [
+      this.penalizerInstance,
+      this.relayRegistrar,
+      this.relayHubInstance,
+      this.stakeManagerInstance].every(it => it == null)
+    if (allNull) {
+      throw new Error('ERC-165 interface check failed. Not a single contract instance initialized')
+    }
+    this.logger.debug(`ERC-165 interface IDs: ${JSON.stringify(erc165Interfaces)}`)
+    const pn = await this._trySupportsInterface(this.penalizerInstance, erc165Interfaces.penalizer)
+    const rr = await this._trySupportsInterface(this.relayRegistrar, erc165Interfaces.relayRegistrar)
+    const rh = await this._trySupportsInterface(this.relayHubInstance, erc165Interfaces.relayHub)
+    const sm = await this._trySupportsInterface(this.stakeManagerInstance, erc165Interfaces.stakeManager)
+    const all = pn && rr && rh && sm
+    if (!all) {
+      throw new Error(`ERC-165 interface check failed. PN: ${pn} RR: ${rr} RH: ${rh} SM: ${sm}`)
+    }
+  }
+
+  async _validateERC165InterfacesClient (): Promise<void> {
+    const allNull = [
+      this.forwarderInstance,
+      this.paymasterInstance].every(it => it == null)
+    if (allNull) {
+      throw new Error('ERC-165 interface check failed. Not a single contract instance initialized')
+    }
+    this.logger.debug(`ERC-165 interface IDs: ${JSON.stringify(erc165Interfaces)}`)
+    const fw = await this._trySupportsInterface(this.forwarderInstance, erc165Interfaces.forwarder)
+    const pm = await this._trySupportsInterface(this.paymasterInstance, erc165Interfaces.paymaster)
+    const all = fw && pm
+    if (!all) {
+      throw new Error(`ERC-165 interface check failed. FW: ${fw} PM: ${pm}`)
+    }
+  }
+
+  private async _trySupportsInterface (
+    contractInstance: IERC165Instance | null,
+    interfaceId: PrefixedHexString): Promise<boolean> {
+    if (contractInstance == null) {
+      return true
+    }
+    try {
+      return await contractInstance.supportsInterface(interfaceId)
```

Comment:
if this call is 
```js
if ( ! await contractInstance.supportsInterface(interfaceId) ) {
  throw new Error( `${name} failed supportInterface(${interfaceId})` )
}
```

then all you need is sequence of calls to `_trySupportsInterface`, without all the funny boolean checks.
(just with added "name" parameter, so the error can include the contract name, and not only address)